pkg/proxy/proxy.go

```diff
@@ -1,267 +0,0 @@
-func (n krpAuthorizerAttributesGetter) GetRequestAttributes(u user.Info, r *http.Request) []authorizer.Attributes {
-	apiVerb := ""
-	switch r.Method {
-	case "POST":
-		apiVerb = "create"
-	case "GET":
-		apiVerb = "get"
-	case "PUT":
-		apiVerb = "update"
-	case "PATCH":
-		apiVerb = "patch"
-	case "DELETE":
-		apiVerb = "delete"
-	}
-
-	var allAttrs []authorizer.Attributes
-
-	defer func() {
-		for attrs := range allAttrs {
-			klog.V(5).Infof("kube-rbac-proxy request attributes: attrs=%#+v", attrs)
-		}
-	}()
-
-	if n.authzConfig.ResourceAttributes == nil {
-		// Default attributes mirror the API attributes that would allow this access to kube-rbac-proxy
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       "",
-			APIGroup:        "",
-			APIVersion:      "",
-			Resource:        "",
-			Subresource:     "",
-			Name:            "",
-			ResourceRequest: false,
-			Path:            r.URL.Path,
-		})
-		return allAttrs
-	}
-
-	if n.authzConfig.Rewrites == nil {
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       n.authzConfig.ResourceAttributes.Namespace,
-			APIGroup:        n.authzConfig.ResourceAttributes.APIGroup,
-			APIVersion:      n.authzConfig.ResourceAttributes.APIVersion,
-			Resource:        n.authzConfig.ResourceAttributes.Resource,
-			Subresource:     n.authzConfig.ResourceAttributes.Subresource,
-			Name:            n.authzConfig.ResourceAttributes.Name,
-			ResourceRequest: true,
-		})
-		return allAttrs
-	}
-
-	params := []string{}
-	if n.authzConfig.Rewrites.ByQueryParameter != nil && n.authzConfig.Rewrites.ByQueryParameter.Name != "" {
-		if ps, ok := r.URL.Query()[n.authzConfig.Rewrites.ByQueryParameter.Name]; ok {
-			params = append(params, ps...)
-		}
-	}
-	if n.authzConfig.Rewrites.ByHTTPHeader != nil && n.authzConfig.Rewrites.ByHTTPHeader.Name != "" {
-		if p := r.Header.Get(n.authzConfig.Rewrites.ByHTTPHeader.Name); p != "" {
-			params = append(params, p)
-		}
-	}
-
-	if len(params) == 0 {
-		return allAttrs
-	}
-
-	for _, param := range params {
-		attrs := authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       templateWithValue(n.authzConfig.ResourceAttributes.Namespace, param),
-			APIGroup:        templateWithValue(n.authzConfig.ResourceAttributes.APIGroup, param),
-			APIVersion:      templateWithValue(n.authzConfig.ResourceAttributes.APIVersion, param),
-			Resource:        templateWithValue(n.authzConfig.ResourceAttributes.Resource, param),
-			Subresource:     templateWithValue(n.authzConfig.ResourceAttributes.Subresource, param),
-			Name:            templateWithValue(n.authzConfig.ResourceAttributes.Name, param),
-			ResourceRequest: true,
-		}
-		allAttrs = append(allAttrs, attrs)
-	}
-	return allAttrs
-}
-
-// DeepCopy of Proxy Configuration
-func (c *Config) DeepCopy() *Config {
```

Comment:
TODO: pre-acceptance either remove or fuzz test. This is similar to the deep copy of rest.config in k8s.io/client-go to be sure that adding fields doesn't result in non-deepcopiable results or incorrect deep copies.

Reply:
let's remove, it seems unused.